disk/src/com/treode/disk/edit/LogReader.scala

```diff
@@ -31,13 +34,19 @@ class LogReader (file: File, geom: DriveGeometry) (implicit scheduler: Scheduler
     var continue = true
     scheduler.whilst (continue) {
       for {
+        // get length of log
         _ <- file.fill (buf, pos, 4)
         length = buf.readInt()
-        _ <- file.fill (buf, pos+4, length)
+        // get batch length, then read `length` bytes
+        _ <- file.fill (buf, pos+4, 4)
+        batch_len = buf.readInt()
+        _ <- file.fill (buf, pos+8, length)
```

Comment:
I found `length` and `batch_len` confusing. After reviewing, I realize one is the length in bytes and the other the length in records. Maybe call one `byteLength` and the other `recordCount`.  Or better yet (IMO), just call them `length` and `count`, and use comments to explain further.

Instead of

    _ <- file.fill (buf, pos, 4)
    length = buf.readInt()
    _ <- file.fill (buf, pos+4, 4)
    count = buf.readInt()
    _ <- file.fill (buf, pos+8, length)

you can do

    _ <- file.fill (buf, pos, 8)
    length = buf.readInt()
    count = buf.readInt()
    _ <- file.fill (buf, pos+8, length)

That reduces the number of calls to file.fill by one. That's a good thing, because you should think of async calls as sortof expensive. Not so expensive that you avoid them at all costs, just expensive enough that you use them wisely.